src/cargo/ops/fix.rs

```diff
@@ -48,9 +50,14 @@ pub fn fix(ws: &Workspace, opts: &mut FixOptions) -> CargoResult<()> {
         opts.compile_opts.build_config.extra_rustc_env.push((key, "1".to_string()));
     }
 
-    if let Some(edition) = opts.edition {
+    if opts.edition {
+        let key = EDITION_ENV.to_string();
+        opts.compile_opts.build_config.extra_rustc_env.push((key, "1".to_string()));
+    }
+
+    if let Some(edition) = opts.prepare_for {
```

Comment:
`else if let` here to only ever use one of edition/prepare_for?

Alternatively, use clap's `conflicts_with("prepare_for")` on the edition arg